src/main/java/ca/on/oicr/gsi/somnus/Server.java

```diff
@@ -157,95 +164,79 @@ protected void renderContent(XMLStreamWriter writer) throws XMLStreamException {
+                    os.write("Creator and services are required".getBytes(StandardCharsets.UTF_8));
+                  }
+                  return;
+                }
+                knownServices.addAll(query.getServices());
+                knownEnvironments.add(query.getEnvironment());
+                final Inhibition inhibition =
+                    new Inhibition(
+                        new TreeSet<>(query.getServices()),
+                        query.getEnvironment(),
+                        Instant.now().plusSeconds(query.getTtl()),
+                        query.getCreator(),
+                        query.getReason());
+                inhibitions.add(inhibition);
+                refreshPrometheus();
+                final CreateResponse response = new CreateResponse();
+                response.setId(inhibition.id());
+                response.setExpirationTime(inhibition.expirationTime().getEpochSecond());
+                t.getResponseHeaders().set("Content-type", "application/json");
+                t.sendResponseHeaders(201, 0);
+                try (final OutputStream os = t.getResponseBody()) {
+                  os.write(mapper.writeValueAsString(response).getBytes(StandardCharsets.UTF_8));
+                }
+              } catch (final Exception e) {
+                e.printStackTrace();
+                t.sendResponseHeaders(400, 0);
+                try (final OutputStream os = t.getResponseBody()) {
+                  os.write("{\"error\":\"Bad JSON\"}".getBytes(StandardCharsets.UTF_8));
+                }
+                return;
               }
-              refreshPrometheus();
-              return;
-            }
 
-            t.sendResponseHeaders(404, 0);
-            try (final OutputStream os = t.getResponseBody()) {
-              os.write("Not found".getBytes(StandardCharsets.UTF_8));
-            }
-          } catch (final Exception e) {
-            e.printStackTrace();
-            t.sendResponseHeaders(400, 0);
-            try (OutputStream os = t.getResponseBody()) {
-              os.write("Bad JSON".getBytes(StandardCharsets.UTF_8));
-            }
-          }
-        });
-    add(
-        "/awakeui",
-        t -> {
-          boolean deleted = false;
-          int id = Integer.parseInt(t.getRequestURI().getQuery());
-          for (final Inhibition inhibition : inhibitions) {
-            if (inhibition.id() != id) continue;
-            inhibition.awake();
-            deleted = true;
-            refreshPrometheus();
-            break;
-          }
-          t.getResponseHeaders().set("Location", "/");
-          t.sendResponseHeaders(302, 0);
-          try (OutputStream os = t.getResponseBody()) {
-            os.write((deleted ? "Deleted" : "Not found").getBytes(StandardCharsets.UTF_8));
+              break;
+            case "DELETE":
+              try {
+                final int id = mapper.readValue(t.getRequestBody(), int.class);
```

Comment:
Strange that the request body isn't JSON. Use a request parameter instead?

Reply:
The request body is JSON, just a single number rather than a number. There's no utilties for parsing request parameters with this server, so it's much easier to take it as the body rather than writing my own.

Reply:
:+1: I thought valid JSON had to be an object or array, but you're right - a single value is valid too: https://tools.ietf.org/html/rfc7159